Easy/TwoIntegerSum/solution.js

```diff
@@ -0,0 +1,44 @@
+class Solution {
+    /**
+     * Solução
+     *  - Primeiro passo: Criei um hashmap para armazenar informações de cada elemento do array passado
+     *      valor = complemento(um valor que somado com o do elemento resulta no target).
+     *      chave = posição do elemento dentro do array nums.
+     * 
+     *  - Segundo passo: Criei um loop para pega um elemento do array nums, calcula seu complemento e adicionar
+     *      o valor e chave dentro do hashmap.
+     * 
+     *  - Terceiro passo: Fiz uma verificação dentro do loop, se seu complemento se encontra dentro do hashmap é retornado o elemento e
+     *      seu complemento.No entanto, caso seja encontrado o complemento é retornado um array vazio.
+     * 
+     *  Big O Espaço: O(n)
+     *  Big O Tempo: O(n)
+     * 
+     *  Tempo necessário para solução: 2h
+     * 
+     *  Precisei pesquisar sobre Map.has(): https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Map/has
+     * 
+     * 
+     * @param {number[]} nums = array de numeros passado.
+     * @param {number} target = valor da soma de 2 elementos dentro de nums. 
+     * @return {number[]} = sempre irá retorna um array de 2 elementos ou vazio.
+     */
+    twoSum(nums, target) {
+        const map = new Map();
+
+        for (let i = 0; i < nums.length; ++i) {
+            const num = nums[i];
+            const complement = target - num;
+            const sumIndex = map.get(complement);
+
+            const isTarget = map.has(complement);
+            if (isTarget) {
+                return [i, sumIndex];
+            }
+
+            map.set(num, i);
+        }
+
+        return [];
+    }
```

Comment:
Algumas sugestões no código:

1. o `sumIndex` é apenas usado para quando o if for verdadeiro, não faz sentido criar ele sem saber se vai usar, isso é um uso errado de declaração de variável, dependendo da linguagem pode até causar um memory leak, já que não se sabe se o `complement` existe dentro de `map`. Sem falar que ela é usada em apenas um lugar.

2. é criado a const `isTarget` para verificar se há o número esperado, primeiro que eu acho que deveria se chamar `hasTarget` no sentindo que o map tem algo (has something). Segundo que essa variável é utilizada apenas em um momento, então não vejo tanto sentido de criar ela.

3. O array `nums` ja tem todos os valores esperados, criar uma variável como o `num` nesse caso não tem muita relevancia, pois ela só está "extraindo" o numero atual, e criando mais uma variável de armazenamento. Como o array ja tem o valor, acredito que seria melhor usar o `nums[i]` direto.

```suggestion
    twoSum(nums, target) {
        const map = new Map();

        for (let i = 0; i < nums.length; ++i) {
            const complement = target - nums[i];

            if (map.has(complement)) {
                return [i, map.get(complement)];
            }

            map.set(nums[i], i);
        }

        return [];
    }
```

Essa é minha opinião sobre o código e como eu escreveria ele, só passando adiante o motivo e como ele ficaria 😅 

Gostei do padrão da escrita do código, espaços certinhos, todos os `;`, quebra de linha antes do `return` 👏👏👏